pkg/controller/awsfederatedrole/awsfederatedrole_controller.go

```diff
@@ -141,6 +141,9 @@ func (r *ReconcileAWSFederatedRole) Reconcile(request reconcile.Request) (reconc
 		PolicyName:     &instance.Spec.AWSCustomPolicy.Name,
 		PolicyDocument: &jsonPolicy,
 	})
+
+	err = awsfaa.CreateOrUpdateIAMPolicy(awsClient, *instance)
+
 	if err != nil {
 		if aerr, ok := err.(awserr.Error); ok {
 			if aerr.Code() == "MalformedPolicyDocument" {
```

Comment:
These functions errors here need to be refactored as well. since you are now not returning an aws error possibly from CreateOrUpdateIAMPolicy we need better handling here for instance 164 still does not update the status of the FAR